File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/provider/RealContentProvider.java
package com.raizlabs.android.dbflow.test.provider;

import android.content.ContentProvider;
import android.content.ContentValues;
import android.database.Cursor;
import android.net.Uri;

import com.raizlabs.android.dbflow.config.FlowManager;
import com.raizlabs.android.dbflow.structure.database.DatabaseWrapper;
import com.raizlabs.android.dbflow.test.TestDatabase;

/**
 * Description:
 */
public class RealContentProvider extends ContentProvider {

    String databaseName = TestDatabase.NAME;

    DatabaseWrapper database;

    @Override
    public boolean onCreate() {
        FlowManager.init(getContext());
        database = FlowManager.getDatabase(databaseName).getWritableDatabase();
        return true;
    }

    @Override
    public Cursor query(Uri uri, String[] projection, String selection, String[] selectionArgs, String sortOrder) {
        return null;
    }

    @Override
    public String getType(Uri uri) {
        return null;
    }

    @Override
    public Uri insert(Uri uri, ContentValues values) {
        return null;
    }

    @Override
    public int delete(Uri uri, String selection, String[] selectionArgs) {
        return 0;
    }

    @Override
    public int update(Uri uri, ContentValues values, String selection, String[] selectionArgs) {
        return 0;
    }
}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/provider/RealContentProvider.kt
package com.raizlabs.android.dbflow.test.provider

import android.content.ContentProvider
import android.content.ContentValues
import android.database.Cursor
import android.net.Uri

import com.raizlabs.android.dbflow.config.FlowConfig
import com.raizlabs.android.dbflow.config.FlowManager
import com.raizlabs.android.dbflow.structure.database.DatabaseWrapper
import com.raizlabs.android.dbflow.test.TestDatabase

/**
 * Description:
 */
class RealContentProvider : ContentProvider() {

    var databaseName = TestDatabase.NAME

    lateinit var database: DatabaseWrapper

    override fun onCreate(): Boolean {
        FlowManager.init(FlowConfig.Builder(context).build())
        database = FlowManager.getDatabase(databaseName).writableDatabase
        return true
    }

    override fun query(uri: Uri, projection: Array<String>?, selection: String?, selectionArgs: Array<String>?, sortOrder: String?): Cursor? {
        return null
    }

    override fun getType(uri: Uri): String? {
        return null
    }

    override fun insert(uri: Uri, values: ContentValues?): Uri? {
        return null
    }

    override fun delete(uri: Uri, selection: String?, selectionArgs: Array<String>?): Int {
        return 0
    }

    override fun update(uri: Uri, values: ContentValues?, selection: String?, selectionArgs: Array<String>?): Int {
        return 0
    }
}
